proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -4,26 +4,54 @@ package determined.experiment.v1;
 option go_package = "github.com/determined-ai/determined/proto/pkg/experimentv1";
 
 import "google/protobuf/struct.proto";
+import "protoc-gen-swagger/options/annotations.proto";
 
-// WorkloadKind defines the kind of workload that should be executed by trial
-// runners.
-enum Unit {
-  // Denotes an unknown Uz.
-  UNIT_UNSPECIFIED = 0;
-  // Unit type "Batches".
-  UNIT_BATCHES = 1;
-  // Unit type "Records".
-  UNIT_RECORDS = 2;
-  // Unit type "Epochs".
-  UNIT_EPOCHS = 3;
+// The representation for an amount of training.
+message TrainingLength {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "units", "length" ] }
+  };
+  // The units for the training length.
+  enum Units {
+    // Zero-value (not allowed).
+    UNITS_UNSPECIFIED = 0;
+    // Indicates a training length is specified in records, samples or another
+    // synonymous unit.
+    UNITS_RECORDS = 1;
+    // Indicates a training length is specified in batches.
+    UNITS_BATCHES = 2;
+    // Indicates a training length is specified in epochs.
+    UNITS_EPOCHS = 3;
+  }
+  // The units the training length is in terms of.
+  Units units = 1;
+  // The value for the training length.
+  int32 length = 2;
+}
+
+// ValidateAfterOperation means the trial should train and validate after
+// training the given length.
+message ValidateAfterOperation {
+  // The length to train before reporting a validation.
+  TrainingLength length = 1;
+}
+
+// Used to complete a ValidateAfterOperation.
+message CompleteValidateAfterOperation {
+  // The ValidateAfterOperation being completed.
+  ValidateAfterOperation op = 1;
+  // The value of searcher metric associated with this completed metric.
+  // The metric provided should be the metric used to guide HP search.
+  double searcher_metric = 2;
 }
 
-// TrainingUnits represents a single runnable operation emitted by a searcher.
-message TrainingUnits {
-  // This is the unit the TrainingLength is in terms of.
-  Unit unit = 1;
-  // This is the magnitude of the TrainingLength.
-  int32 count = 2;
+// SearcherOperation is an operation issued by the searcher.
+message SearcherOperation {
+  // A searcher operation is one of the following operations.
+  oneof union {
```

Comment:
Wait, why is this a union type?

Reply:
backwards compatability, in the event we ever add more operations (pls no but you know, idk the direction hpsearch is moving in), then it won't be a breaking change.